### PR TITLE
Fix failover test for reparenting

### PR DESCRIPTION
**Pull request by dbussink**
This test is pretty broken, as it uses MariaDB replication, but we're running this with MySQL.

The reason the test doesn't outright fail is a combination of two things. First, the test doesn't assert that the command to change replication don't error, so it returns an error.

Secondly, this is also affect by the multi execute issue of #14948, so it's not even guaranteed the queries run properly.

Marked for backporting as this test is broken also in older releases and can lead to test flakiness and lack of exercising the right thing to avoid future regressions. 

## Related Issue(s)

Part of #14948

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by shlomi-noach — APPROVED** [review]

File: go/test/endtoend/tabletgateway/buffer/reparent/failover_buffer_test.go [hunk 1]
```diff
@@ -62,16 +65,27 @@ func failoverExternalReparenting(t *testing.T, clusterInstance *cluster.LocalPro
...
 
 	// Use 'localhost' as hostname because Travis CI worker hostnames
 	// are too long for MySQL replication.
-	changeSourceCommands := fmt.Sprintf("RESET SLAVE;SET GLOBAL gtid_slave_pos = '%s';CHANGE MASTER TO MASTER_HOST='%s', MASTER_PORT=%d ,MASTER_USER='vt_repl', MASTER_USE_GTID = slave_pos;START SLAVE;", gtID, "localhost", newPrimary.MySQLPort)
-	oldPrimary.VttabletProcess.QueryTablet(changeSourceCommands, keyspaceUnshardedName, true)
+	changeSourceCommands := []string{
+		"STOP SLAVE",
+		"RESET MASTER",
+		fmt.Sprintf("SET GLOBAL gtid_purged = '%s'", gtID),
+		fmt.Sprintf("CHANGE MASTER TO MASTER_HOST='%s', MASTER_PORT=%d, MASTER_USER='vt_repl', MASTER_AUTO_POSITION = 1", "localhost", newPrimary.MySQLPort),
```
Comment (shlomi-noach) [inline go/test/endtoend/tabletgateway/buffer/reparent/failover_buffer_test.go:83]: Missing a `START SLAVE` command.

Comment (dbussink) [reply go/test/endtoend/tabletgateway/buffer/reparent/failover_buffer_test.go:83]: Fixed!

**Comment by shlomi-noach** [comment]
Incorporated into https://github.com/vitessio/vitess/pull/14949

**Comment by dbussink** [comment]
> Incorporated into #14949

Closing in favor of this one.

**Pull request closed without merging**